src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -129,7 +129,18 @@ public ResettableSqlStatement GetResettableStatement(string query)
             return new ResettableSqlStatement(statement);
         }
 
-        public void RunInTransaction(Action action)
+        public void RunInTransaction<TState>(Action<TState> action, TState state)
+        {
+            RunInTransaction(
+                state =>
```

Comment:
fwiw, calling this state was super confusing.

Reply:
😕 I thought that was the normal name for this. Examples of alternatives?

Reply:
it's more that the outer param for RunInTransaction is called "state" and hte inner one is called "state" and my brain exploded with confusion.  Honestly, i'd be 100% fine with you calling the inner one `tuple` :)

Reply:
it's also a nit.  if you're ok with this, it's NBD.  just giving you my stream of consciousness thoughts as i tried to grok things :)

Reply:
I've seen other instances of this pattern with `TArg` and `arg`.

Reply:
Note: the pattern is fine.  it's literally the fact that there are two params called 'state', and then the code has `state.state` :D  It's purely naming i don't like.  `arg` or `tuple` would be great IMO just for some simple clarity.